src/game/common/recorder.h

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Recorder
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+#include "always.h"
+#include "gameinfo.h"
+#include "subsysteminterface.h"
+
+class CRCInfo;
+
+enum RecorderModeType
+{
+    RECORDERMODETYPE_RECORD,
+    RECORDERMODETYPE_PLAYBACK,
+};
+
+class ReplayGameInfo : public GameInfo
+{
+private:
+    GameSlot m_ReplaySlot[8];
+};
+
+class RecorderClass : public SubsystemInterface
+{
+public:
+    virtual ~RecorderClass() override;
+    virtual void Init() override;
+    virtual void Reset() override;
+    virtual void Update() override;
+
+private:
+    CRCInfo *m_crcInfo;
+    FILE *m_file;
+    Utf8String m_fileName;
+    int m_currentFilePosition;
+    RecorderModeType m_mode;
+    Utf8String m_unk;
+    ReplayGameInfo m_gameInfo;
+    bool m_wasDesync;
+    bool m_doLogging;
```

Comment:
i investigated this a while back, doesn't have anything to do with logging, not sure what its for, class clears map if not set

Reply:
Renamed it.